File: sample-extension/src/main/java/com/airsaid/sample/extension/component/code/SampleSourceCode.java
package com.airsaid.sample.extension.component.code;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author JackChen
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface SampleSourceCode {
  /**
   * If you want to filter the source files by regex.
   */
  String regex() default "";
}

File: sample-extension/src/main/java/com/airsaid/sample/extension/component/code/SampleSourceCode.kt
package com.airsaid.sample.extension.component.code

/**
 * @author airsaid
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class SampleSourceCode(
  /**
   * If you want to filter the source files by regex.
   */
  val regex: String = "",
)
